src/Datadog.Trace/TraceId.cs

```diff
@@ -0,0 +1,155 @@
+using System;
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// Class representing 64 or 128 bit TraceID.
+    /// </summary>
+    public readonly struct TraceId
```

Comment:
Can you please describe (e.g. in PR description) the decisions that were behind this design? I think you could even try to describe the whole design process which resulted in having this implementation of `TraceId`

I think that for backing fields we can have:
- higher + lower `ulong` (`lower == 0` means 64 bit, the collision probability is low - probably collision would not make any big harm)
- `[]byte`
- hex formatted `string`

Have you compared your implementation with https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L1747? It uses hex formatted `string` as backing field and I think it makes sense as this is the usual input and output for tracing (some headers).

Reply:
TBH I did not think to add ALL possible backing fields 😄 
I looks like the `byte[]` is used only in context propagation so probably this one at least could be removed. And maybe changed to some extension method like `AsBytes()`.